server/src/main/java/org/opensearch/search/pipeline/SearchPipelineService.java

```diff
@@ -366,6 +370,23 @@ public SearchResponse transformResponse(SearchRequest request, SearchResponse se
         return searchResponse;
     }
 
+    public <Result extends SearchPhaseResult> SearchPhaseResults<Result> transformSearchPhase(
```

Comment:
I'm thinking about how this will merge with https://github.com/opensearch-project/OpenSearch/pull/7253 (where the search pipeline may be defined in the request source). As part of that change, I replaced `transformRequest` and `transformResponse` on SearchPipelineService with a `resolvePipeline` method. Then the caller uses the `Pipeline` to process the request/response without calling into SearchPipelineService again.

Maybe we should add a nullable `Pipeline` member to `SearchPhaseContext`?

Reply:
@msfroh That is not required. SearchPhaseContext contains a request object. I can use that request object to resolve the pipeline. 

Ref: https://github.com/opensearch-project/OpenSearch/blob/7945f01055a6770eb62bf5e0d82623d8fc302923/server/src/main/java/org/opensearch/action/search/SearchPhaseContext.java#L74

Reply:
Thats a fair point. If put the pipeline in the search request then we just resolved it once.

Reply:
As a bonus, if the `SearchPhaseContext` holds a reference to the `Pipeline`, we won't need to pass `SearchPipelineService` to all of the `Search*Action` classes.